presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+    @ScalarFunction("ST_LineFromText")
+    @SqlType(NAME)
+    public static Slice stLineString(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "ST_LineFromText", LINE_STRING);
+        return serialize(geometry);
+    }
+
+    @Description("Returns a Geometry type Point object with the given coordinate values")
+    @ScalarFunction("ST_Point")
+    @SqlType(NAME)
+    public static Slice stPoint(@SqlType(StandardTypes.DOUBLE) double x, @SqlType(StandardTypes.DOUBLE) double y)
+    {
+        OGCGeometry geometry = OGCGeometry.createFromEsriGeometry(new Point(x, y), null);
+        return serialize(geometry);
+    }
+
+    @Description("Returns a Geometry type Polygon object from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_Polygon")
+    @SqlType(NAME)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "ST_Polygon", POLYGON);
+        return serialize(geometry);
+    }
+
+    @Description("Returns the area of a polygon using Euclidean measurement on a 2D plane")
+    @ScalarFunction("ST_Area")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stArea(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Area", POLYGON);
+        return geometry.getEsriGeometry().calculateArea2D();
+    }
+
+    @Description("Returns a Geometry type object from Well-Known Text representation (WKT)")
+    @ScalarFunction("ST_GeometryFromText")
+    @SqlType(NAME)
+    public static Slice stGeometryFromText(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(OGCGeometry.fromText(input.toStringUtf8()));
+    }
+
+    @SqlNullable
+    @Description("Returns the Well-Known Text (WKT) representation of the geometry")
+    @ScalarFunction("ST_AsText")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stAsText(@SqlType(NAME) Slice input)
+    {
+        return Slices.utf8Slice(deserialize(input).asText());
+    }
+
+    @Description("Returns the Point value that is the mathematical centroid of the Polygon")
+    @ScalarFunction("ST_Centroid")
+    @SqlType(NAME)
+    public static Slice stCentroid(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Centroid", POLYGON);
+        Polygon polygon = (Polygon) geometry.getEsriGeometry();
+        int pointCount = polygon.getPointCount();
+        double xCoordinate = 0;
+        double yCoordinate = 0;
+        for (int i = 0; i < pointCount; i++) {
+            xCoordinate += polygon.getPoint(i).getX();
+            yCoordinate += polygon.getPoint(i).getY();
+        }
+        Point centroid = new Point(xCoordinate / pointCount, yCoordinate / pointCount);
```

Comment:
According to https://en.wikipedia.org/wiki/Centroid this implementation calculates centroid of a set of points and not centroid of a polygon. The formula for the latter is more complex and includes an area of the polygon.

Also, for an polygon "POLYGON EMPTY", will pointCount be 0 and will it cause a division-by-zero error?